packages/push/src/PushRegistration.ts

```diff
@@ -15,15 +19,85 @@ declare var window: any;
 export class PushRegistration {
 
   public static readonly TYPE: string = "push";
-  public static readonly API_PATH: string = "rest/registry/device";
 
-  private readonly pushConfig?: ServiceConfiguration;
-  private _objectInstance: any;
+  private static readonly API_PATH: string = "rest/registry/device";
+  private static readonly TOKEN_KEY = "UPS_DEVICE_TOKEN";
+
+  private readonly validationError?: string;
+  private readonly variantId?: string;
+  private readonly http?: AxiosInstance;
+  private readonly push?: any;
 
   constructor(config: ConfigurationService) {
     const configuration = config.getConfigByType(PushRegistration.TYPE);
+
     if (configuration && configuration.length > 0) {
-      this.pushConfig = configuration[0];
+      const pushConfig = configuration[0];
+
+      if (!window || !window.device || !window.PushNotification) {
+        const errorMessage = "@aerogear/cordova-plugin-aerogear-push plugin not installed.";
+        this.validationError = errorMessage;
+        console.warn(errorMessage);
+      }
+
+      if (!pushConfig || !pushConfig.config) {
+        const errorMessage = "UnifiedPush server configuration not found";
+        this.validationError = errorMessage;
+        console.warn(errorMessage);
+      }
+
+      const unifiedPushServerURL = pushConfig.url;
+      if (!pushConfig.url) {
+        const errorMessage = "UnifiedPush server URL not found";
+        this.validationError = errorMessage;
+        console.warn(errorMessage);
+      }
+
+      let platformConfig;
+      if (isCordovaAndroid()) {
+        platformConfig = pushConfig.config.android;
+      } else if (isCordovaIOS()) {
+        platformConfig = pushConfig.config.ios;
+      } else {
+        const errorMessage = "Platform is not supported by UnifiedPush";
+        this.validationError = errorMessage;
+        console.warn(errorMessage);
+      }
+
+      this.variantId = platformConfig.variantId || platformConfig.variantID;
+      if (!this.variantId) {
+        const errorMessage = "UnifiedPush VariantId is not defined";
+        this.validationError = errorMessage;
+        console.warn(errorMessage);
+      }
+
+      const variantSecret = platformConfig.variantSecret;
+      if (!variantSecret) {
+        const errorMessage = "UnifiedPush VariantSecret is not defined";
+        this.validationError = errorMessage;
+        console.warn(errorMessage);
+      }
+
+      if (!this.validationError) {
+        const token = window.btoa(`${this.variantId}:${variantSecret}`);
+        this.http = axios.create({
+          baseURL: unifiedPushServerURL,
+          timeout: 5000,
+          headers: {"Authorization": `Basic ${token}`}
+        });
+      }
+
+      this.push = window.PushNotification.init(
```

Comment:
This I think this should probably stay in the `register` method right?

Reply:
In the PR maybe but... I will need this in unregister to unsubscribe the device from the Firebase topics (next PR) so that is why I have move it to the constructor.

Reply:
I'm not sure what you mean. Do you mean you will need to access `this.push` from within the `unregister` method? That should be available because ideally the `register` method will have been called already before calling `unregister` right? And if you call `unregister` without having called `register` I would expect an error right?

From what I can see in the constructor, we are now guaranteed to always call `window.PushNotification.init()` even if `window.PushNotification` is undefined. This is a change in business logic because previously we only called it in the `register` method and only if `window.PushNotification` was defined. I think this could potentially cause some crashes if someone accidentally has a push config in their app while they run it in the browser.

If you think it's absolutely necessary to have it in the constructor it should at least be surrounded by the `if (!window || !window.device || !window.PushNotification)` statement on Line 37.

Also, we need unit tests.

Reply:
> I'm not sure what you mean. Do you mean you will need to access this.push from within the unregister method? That should be available because ideally the register method will have been called already before calling unregister right? And if you call unregister without having called register I would expect an error right?

You can call register and than in another time need to call unregister but don't have the `PushRegistration` instance anymore, that is why I have added it in the construtor.

> From what I can see in the constructor, we are now guaranteed to always call window.PushNotification.init() even if window.PushNotification is undefined. This is a change in business logic because previously we only called it in the register method and only if window.PushNotification was defined. I think this could potentially cause some crashes if someone accidentally has a push config in their app while they run it in the browser.

Nice catch, I'm going to move it inside of `if (!this.validationError) {` block, because in that point we already had check `window.PushNotification`